src/providers/staticfile.rs

```diff
@@ -0,0 +1,117 @@
+use super::Provider;
+use crate::nixpacks::{
+    app::{App, StaticAssets},
+    environment::Environment,
+    nix::pkg::Pkg,
+    phase::{BuildPhase, SetupPhase, StartPhase},
+};
+use anyhow::Result;
+use indoc::formatdoc;
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+
+#[derive(Serialize, Deserialize, Default, Debug)]
+pub struct Staticfile {
+    pub root: Option<String>,
+    pub directory: Option<String>,
+    pub gzip: Option<String>,
+    pub status_code: Option<HashMap<u32, String>>,
+}
+
+pub struct StaticfileProvider {}
+
+impl Provider for StaticfileProvider {
+    fn name(&self) -> &str {
+        "staticfile"
+    }
+
+    fn detect(&self, app: &App, _env: &Environment) -> Result<bool> {
+        Ok(app.includes_file("Staticfile"))
+    }
+
+    fn setup(&self, _app: &App, _env: &Environment) -> Result<Option<SetupPhase>> {
+        let pkg = Pkg::new("nginx");
+        Ok(Some(SetupPhase::new(vec![pkg])))
+    }
+
+    fn static_assets(&self, app: &App, _env: &Environment) -> Result<Option<StaticAssets>> {
+        let mut assets = StaticAssets::new();
+
+        let mut mime_types = "".to_string();
+        if app.includes_file("mime.types") {
+            assets.insert("mime.types".to_string(), app.read_file("mime.types")?);
+            mime_types = "include\tmime.types;".to_string();
+        }
+
+        let mut auth_basic = "".to_string();
+        if app.includes_file("Staticfile.auth") {
+            assets.insert(".htpasswd".to_string(), app.read_file("Staticfile.auth")?);
+            auth_basic = format!(
+                "auth_basic\t\"Password Required\";\nauth_basic_user_file\t{};",
+                app.asset_path(".htpasswd")
+            );
+        }
+
+        let staticfile: Staticfile = app.read_yaml("Staticfile").unwrap_or_default();
+        let port = std::env::var("PORT").unwrap_or_else(|_| "80".to_string());
```

Comment:
environment variable reading should come from the `Environment` struct, not from reading the variable directly.